core/code/ornaments.js

```diff
@@ -71,5 +139,109 @@ window.ornaments = {
+      if (dataStr === undefined) { return; }
+      this.excludedOrnaments = JSON.parse(dataStr);
+    } catch (e) {
+      console.warn('ornaments: failed to load excludedOrnaments from localStorage: '+e);
+    }
+    try {
+      dataStr = localStorage['knownOrnaments'];
+      if (dataStr === undefined) {
+        this.initOrnaments ();
+        return;
+      }
+      this.knownOrnaments = JSON.parse(dataStr);
+    } catch (e) {
+      console.warn('ornaments: failed to load data from localStorage: '+e);
+    }
+
+  },
+
+  save: function () {
+    localStorage['excludedOrnaments'] = JSON.stringify(this.excludedOrnaments);
+    localStorage['knownOrnaments'] = JSON.stringify(this.knownOrnaments);
+  },
+
+  // reload: addPortal also calls removePortal
+  reload: function () {
+    for (var guid in window.ornaments._portals) {
+      window.ornaments.addPortal(window.portals[guid]);
+    }
+  },
+
+  processInput: function () {
+    window.ornaments.excludedOrnaments = $('#ornaments_E').val().split(/[\s,]+/);
+    window.ornaments.excludedOrnaments = window.ornaments.excludedOrnaments.filter(function (ornamentCode) { return ornamentCode !== ''; });
+    // process the input from the checkboxes
+    for (var ornamentCode in window.ornaments.knownOrnaments) {
+      var input = document.getElementById('chk_orn_'+ornamentCode);
+      window.ornaments.knownOrnaments[ornamentCode] = input ? input.checked : false; // <- default value if the input is not found for unexpected reason
+    }
+  },
+
+  ornamentsList: function() {
+    var text ='';
+    var sortedIDs = Object.keys(window.ornaments.knownOrnaments).sort();
+
+    sortedIDs.forEach ( function (ornamentCode) {
+      var hidden = window.ornaments.excludedOrnaments.some( function(code) {
+        return ornamentCode.startsWith(code);
+      });
+
+      var name = (window.ornaments.icon[ornamentCode] ? window.ornaments.icon[ornamentCode].name +' ('+ornamentCode+')' : ornamentCode);
+      var checked = (window.ornaments.knownOrnaments[ornamentCode]||hidden) ?  'checked ' : '';
+      text += '<label><input id="chk_orn_' + ornamentCode + '" type="checkbox" ' + checked;
+      text += ' onchange="window.ornaments.processInput();window.ornaments.save();window.ornaments.reload()"';
+      text += (hidden ? 'disabled':'');
+      text += '>'+ name + '</label><br>';
+    });
+    return text;
+  },
+
+  replaceOL: function () {
+    document.getElementById('ornamentsList').innerHTML = window.ornaments.ornamentsList();
+  },
+
+  ornamentsOpt: function () {
+    var excludedIDs = window.ornaments.excludedOrnaments.toString();
+    var html = '<div class="ornamentsOpts">'
+             + 'Hide Ornaments from IITC that start with:<br>'
+             + '<input type="text" value="' + excludedIDs + '" id="ornaments_E"'
+             + ' onchange="window.ornaments.processInput();window.ornaments.replaceOL();window.ornaments.save();window.ornaments.reload()"></input><br>'
```

Comment:
how about putting these 4 calls in an extra function?

Reply:
implemented